### ISSUE TITLE
Add ability to allow channel / call credentials over insecure connection (localhost)

### ISSUE THREAD

**Issue by saltxwater** [grpc/grpc-dotnet#1338]
### Is your feature request related to a problem? Please describe.
Despite only listening on localhost / connecting to localhost, I'm still not able to supply call / channel credentials over insecure (no TLS).

### Describe the solution you'd like
Either:
- Allow channel / call credentials (metadata) to be sent (and received) when using no TLS if you're connecting to localhost.
- Or add an environment variable to override the current behavior of blocking these if no TLS regardless of where you're connecting.
- Or a combination of both. ie. Still don't send creds when no TLS even when connecting to localhost, unless the env var is set.

### Describe alternatives you've considered
When using no TLS the whole request itself is passed... so I could change ALL of our current request bodies to include another field which will have the credentials inside. This would involve updated hundreds of proto definitions and services and would bloat the requests to get around the current restriction.

### Additional context
I'm migrating to using Consul Connect as our service mesh. Using this, all our services will only be listening on localhost as well as connecting to upstream services also with localhost. These connections will all be insecure (no tls). Consul Connect will run a sidecar envoy proxy to forward the downstream / upstream requests accordingly to other services in the mesh over TLS, managed by consul.

**Comment by BrennanConroy** [comment]
From https://docs.microsoft.com/aspnet/core/grpc/authn-and-authz?view=aspnetcore-5.0#bearer-token-authentication

>Configuring ChannelCredentials on a channel is an alternative way to send the token to the service with gRPC calls. A ChannelCredentials can include CallCredentials, which provide a way to automatically set Metadata.
CallCredentials is run each time a gRPC call is made, which avoids the need to write code in multiple places to pass the token yourself. Note that CallCredentials are only applied if the channel is secured with TLS. CallCredentials aren't applied on unsecured non-TLS channels.

A regular interceptor can set an authorization header without TLS which is recommended for your scenario.

**Comment by saltxwater** [comment]
Thanks for getting back to me. I interpreted the docs to suggest that no metadata headers would be sent when the channel is not secured. I tested this by adding headers to the request and didn't see the expected header on the service.
Having just tried it again after reading your comment, I see now that my header did get sent, the key was just lower cased which is why my quick test didn't pick it up!

Thanks - I think I've got the functionality I need after all!
